src/LogMonitorEntry.js

```diff
@@ -38,18 +50,35 @@ export default class LogMonitorEntry extends Component {
     let errorText = error;
     if (!errorText) {
       try {
+        const data = this.props.select(state);
+        const previousData = typeof this.props.previousState !== 'undefined' ?
+          this.props.select(this.props.previousState) :
+          undefined;
+        const getValueStyle = ({ style }, nodeType, keyPath) => ({
+          style: {
+            ...style,
+            ...(dataIsEqual(data, previousData, keyPath) ? {} : styles.changedData)
+          }
+        });
+        const getNestedNodeStyle = ({ style }, nodeType, expanded, keyPath) => ({
+          style: {
+            ...style,
+            ...(keyPath.length > 1 ? {} : styles.root)
+          }
+        });
+
         return (
           <JSONTree
-            theme={this.props.theme}
+            theme={{
```

Comment:
Does this mean that any rendering optimizations inside `JSONTree` get disabled because the `theme` object is different every time?